lib/libzfs/libzfs_pool.c

```diff
@@ -4160,39 +4162,78 @@ zpool_get_errlog(zpool_handle_t *zhp, nvlist_t **nverrlistp)
 	verify(nvlist_alloc(nverrlistp, 0, KM_SLEEP) == 0);
 
 	/*
-	 * Fill in the nverrlistp with nvlist's of dataset and object numbers.
+	 * Fill in the nverrlistp with nvlist's of dataset number, object number
+	 * level position, and block id.
 	 */
+	nvlist_t *object_nv;
+	uint64_t *object_block_ids = zfs_alloc(zhp->zpool_hdl,
```

Comment:
If this fails due to nomem, I think it would be better if we could try reverting to earlier behaviour (nvlist's of dataset and object numbers)